platforms/jvm/ear/src/main/java/org/gradle/plugins/ear/EarPlugin.java

```diff
@@ -129,22 +128,6 @@ private void setupEarTask(final Project project, PluginContainer plugins) {
     private void wireEarTaskConventions(Project project) {
         project.getTasks().withType(Ear.class).configureEach(task -> {
             task.getAppDirectory().convention(project.provider(() -> project.getLayout().getProjectDirectory().dir("src/main/application")));
-            task.getConventionMapping().map("libDirName", new Callable<String>() {
-                @Override
-                public String call() throws Exception {
-                    return DEFAULT_LIB_DIR_NAME;
-                }
-            });
-            task.getConventionMapping().map("deploymentDescriptor", new Callable<DeploymentDescriptor>() {
-                @Override
-                public DeploymentDescriptor call() throws Exception {
-                    DeploymentDescriptor deploymentDescriptor = objectFactory.newInstance(DefaultDeploymentDescriptor.class);
-                    deploymentDescriptor.readFrom("META-INF/application.xml");
-                    deploymentDescriptor.readFrom("src/main/application/META-INF/" + deploymentDescriptor.getFileName());
-                    return deploymentDescriptor;
-                }
-            });
```

Comment:
:question: It looks to me that this convention mapping was actually never used because the task's `deploymentDescriptor` was already always set in the `setupEarTask` function above, correct?

I'm asking because at first I tought there might be an issue with these changes due to the fact that the convention mapping was lasily executed, reading JEE xml files from disk. I thought that changing this behavior might have side effects. But now I see that we already read them from disk when configuring the task anyway.

Reply:
Yes this is also my understanding, that's why I removed it.